test/testFullfrequency/testFullfrequency.ino

```diff
@@ -0,0 +1,127 @@
+/**
+ *  Test in all frequencies of US_915 band
+ *      Using SUB BAND 1 [ 903.9 - 905.3 Mhz] see config.h
+ *      channel [0 - 7]
+ *      data rate [0 - 3]
+ *  Class A and C devices with ABP activation
+ *      
+ *  Gateway:    Sentrius RG101
+ * 
+ * Using CATWAN USB STICK hardware
+ * 
+ * Author: Ivan Moreno
+ *  May 2019
+ */
+
+#include <lorawan.h>
+
+// ABP device
+const char *devAddr = "00000000";
+const char *nwkSKey = "00000000000000000000000000000000";
+const char *appSKey = "00000000000000000000000000000000";
+
+
+unsigned long interval = 15000;   // 15 s interval to send message
+unsigned long previousMillis = 0; // will store last time message sent
+unsigned long currentMillis;
+
+union _cntCI {
+  unsigned char _cb[2];
+  unsigned int val;
+} counter;
+
+unsigned char payload[10];
+char outStr[100];
+byte recvStatus = 0;
+
+byte _dataRate;
+byte _channel;
+byte _times;
+
+// Tested on USB STICK CATWAN from electronic cats 
+const sRFM_pins RFM_pins = {
+  .CS = SS,
+  .RST = RFM_RST,
+  .DIO0 = RFM_DIO0,
+  .DIO1 = RFM_DIO1,
+  .DIO2 = RFM_DIO2,
+  .DIO5 = RFM_DIO5,
+};
+
+void setup() {
+  // seti serial port
+  Serial.begin(115200);
+  delay(4000);
+
+  if(!lora.init()){
+    Serial.println("RFM95 not detected");
+    return;
+  }
+
+  // Set LoRaWAN Class
+  lora.setDeviceClass(CLASS_A);
+
+  lora.setTxPower(15);
+
+  counter.val = 0;
+  _channel = 0; // 0 - 7
+  _dataRate = 0; // 0 - 3 
+  _times = 0;
+
+  // Put ABP Key and DevAddress here
+  lora.setNwkSKey((char *)nwkSKey);
+  lora.setAppSKey((char *)appSKey);
+  lora.setDevAddr((char *)devAddr);
+}
+
+void loop() {
+  currentMillis = millis();
+  // Check interval overflow
+  if(currentMillis - previousMillis > interval) {
+    if(_times > 0){
+        _times = 0;
+        _dataRate++;
+        if (_dataRate > 3){
+            _dataRate = 0;
+            _channel++;
+            if (_channel > 7) {
+                _channel = 0;
+            } 
+        }
+    }
+    // Set Channel
+    lora.setChannel(_channel);
```

Comment:
In this part the library is expecting a channel_t instead of a integer

Reply:
Arduino version 1.8.9, travis arduino version 1.8.2

Reply:
@wero1414  I forgot modify this example, i'll do it in the next commit
@sabas1080 what do you mean about  version?

Reply:
@ivan28823 I'm sorry, I thought it was the arduino version